src/control/cmd/agent/main.go

```diff
@@ -49,15 +51,58 @@ const (
 	defaultConfigFile = "daos_agent.yml"
 )
 
+type cmdLogger interface {
+	setLog(*logging.LeveledLogger)
+}
+
+type logCmd struct {
+	log *logging.LeveledLogger
+}
+
+func (c *logCmd) setLog(log *logging.LeveledLogger) {
+	c.log = log
+}
+
+type (
+	jsonOutputter interface {
+		enableJsonOutput(bool)
+		jsonOutputEnabled() bool
+		outputJSON(io.Writer, interface{}) error
+	}
+
+	jsonOutputCmd struct {
+		shouldEmitJSON bool
+	}
+)
+
+func (cmd *jsonOutputCmd) enableJsonOutput(emitJson bool) {
+	cmd.shouldEmitJSON = emitJson
+}
+
+func (cmd *jsonOutputCmd) jsonOutputEnabled() bool {
+	return cmd.shouldEmitJSON
+}
+
+func (cmd *jsonOutputCmd) outputJSON(out io.Writer, in interface{}) error {
+	data, err := json.Marshal(in)
+	if err != nil {
+		return err
+	}
+
+	_, err = out.Write(data)
+	return err
+}
+
 type cliOptions struct {
 	AllowProxy bool       `long:"allow-proxy" description:"Allow proxy configuration via environment"`
-	Debug      bool       `short:"d" long:"debug" description:"Enable debug output"`
+	Debug      string     `short:"d" long:"debug" optional:"1" optional-value:"basic" choice:"basic" choice:"net" description:"Enable basic or enhanced network debug"`
```

Comment:
So what happens if I run `daos_agent --debug` after this change? Does it default to "basic"?

Reply:
Yes, it defaults to 'basic' i.e. normal debug.  Kris suggested that, which looked like a reasonable thing to do.

Reply:
Sure, and I agree. Just wanted to check my assumption about how it works.